db.go

```diff
@@ -893,6 +903,60 @@ func (db *DB) IsReadOnly() bool {
 	return db.readOnly
 }
 
+func (db *DB) freepages() []pgid {
+	tx, err := db.beginTx()
+	defer func() {
+		err = tx.Rollback()
+		if err != nil {
+			panic("freepages: failed to rollback tx")
+		}
+	}()
+	if err != nil {
+		panic("freepages: failed to open read only tx")
+	}
+
+	reachable := make(map[pgid]*page)
+	checkReachable(tx, &tx.root, reachable)
+
+	var fids []pgid
+	for i := pgid(2); i < db.meta().pgid; i++ {
+		if _, ok := reachable[i]; !ok {
+			fids = append(fids, i)
+		}
+	}
+	return fids
+}
+
+func checkReachable(tx *Tx, b *Bucket, reachable map[pgid]*page) {
```

Comment:
this is very similar to `Tx.checkBucket`; would it be possible to have this so both share some code? Maybe `func (tx *Tx) freepages() []pgid`?